apps/api-extractor/src/generators/packageTypings/AstDeclaration.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as ts from 'typescript';
+import { AstSymbol } from './AstSymbol';
+import { Span } from '../../utils/Span';
+
+export interface IAstDeclarationParameters {
+  readonly declaration: ts.Declaration;
+  readonly astSymbol: AstSymbol;
+  readonly parentAstDeclaration: AstDeclaration | undefined;
+}
+
+export class AstDeclaration {
+  public readonly declaration: ts.Declaration;
+
+  public readonly astSymbol: AstSymbol;
+
+  /**
+   * The parent, if this object is nested inside another AstDeclaration.
+   */
+  public readonly parent: AstDeclaration | undefined;
+
+  private readonly _analyzedChildren: AstDeclaration[] = [];
+
+  private readonly _analyzedReferencedAstSymbolsSet: Set<AstSymbol> = new Set<AstSymbol>();
+  private readonly _analyzedReferencedAstSymbols: AstSymbol[] = [];
```

Comment:
>ivate readonly _analyzedReferencedAstSymbols: AstSymbol[] = []; [](start = 4, length = 63)

You might be able to just store the set and convert it to an array when you are ready to iterate

[https://stackoverflow.com/questions/20069828/how-to-convert-set-to-array](https://stackoverflow.com/questions/20069828/how-to-convert-set-to-array)

[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from)

Reply:
Great suggestion. I just used the [...set] notation. This works fine now that we're targeting ES6.

---
In reply to: [173999330](https://github.com/Microsoft/web-build-tools/pull/560#discussion_r173999330) [](ancestors = 173999330)